### PR TITLE
chore: update to `Stylelint` v14

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
[WEB-524]

## Test Plan
- Check if style rules are good
- Check if style lint works well

## Commentary (optional)
I followed [this instruction](https://stylelint.io/migration-guide/to-14/) to migrate from v13 to v14.
v14 automatically forced us to use kabab case, but we use camel case for our scc and kabab case from antd, so i disabled these naming rules.
Stylelint v14 introduced new rules by default such as `shorthand-property-no-redundant-values`.  That's why there are some file changes.

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[WEB-524]: https://determinedai.atlassian.net/browse/WEB-524?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by keita-determined** [comment]
Test fails cuz of SaaS. I'll update `stylelint` in SaaS as soon as this PR is approved

File: webui/react/src/components/Markdown.module.scss
Comment (ClaireNeveu) [inline webui/react/src/components/Markdown.module.scss:286]: Didn't CSS Colors 4 just come out this year? What's the browser support for this?

File: webui/react/src/shared/configs/.stylelintrc.js
Comment (ClaireNeveu) [inline webui/react/src/shared/configs/.stylelintrc.js:33]: Please comment on the rules you changed.

File: webui/react/src/shared/styles/antd.scss
Comment (ClaireNeveu) [inline webui/react/src/shared/styles/antd.scss:37]: This seems like a semantic change.

File: webui/react/src/components/Markdown.module.scss
Comment (keita-determined) [reply webui/react/src/components/Markdown.module.scss:286]: The support is the same as `rgba`. ref: https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/rgb

File: webui/react/src/shared/styles/antd.scss
Comment (keita-determined) [reply webui/react/src/shared/styles/antd.scss:37]: let me check

File: webui/react/src/components/Markdown.module.scss
Comment (ClaireNeveu) [reply webui/react/src/components/Markdown.module.scss:286]: I think you're misunderstanding the note there:
> In browsers that support the standard [rgba()](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/rgba) is an alias for rgb()
Which browsers support the Colors 4 standard?

Comment (keita-determined) [reply webui/react/src/components/Markdown.module.scss:286]: I believe this one https://caniuse.com/css-rrggbbaa

File: webui/react/src/shared/styles/antd.scss
Comment (keita-determined) [reply webui/react/src/shared/styles/antd.scss]: I changed a rule https://stylelint.io/user-guide/rules/list/selector-not-notation/ to have backward compatibility, so reverted these changes

**Review by ClaireNeveu — APPROVED** [review]
Looks good to me but please don't merge without @hamidzr's approval since this affects SaaS.

File: webui/react/src/components/ThemeToggle.module.scss
Comment (hamidzr) [inline webui/react/src/components/ThemeToggle.module.scss:33]: 👍🏼 did it automatically do this?

**Review by hamidzr — APPROVED** [review]
🚢 I didn't follow or look up all the rule descriptions that are changed here but the resulting code changes LGTM and since this is a change around our dev tooling saas should be fine as well. 
It's possible to double check saas integration since the automated test is failing early and not helping more than that, by replicating the change on the saas side and making sure the outcome is sane but this seems low risk

File: webui/react/src/components/ThemeToggle.module.scss
Comment (keita-determined) [reply webui/react/src/components/ThemeToggle.module.scss:33]: I changed it by hand but `--fix` option is available

Comment (hamidzr) [reply webui/react/src/components/ThemeToggle.module.scss:33]: that reminds me in saas I needed to add stylelink as part of `make fmt` same as we did to eslint (it looks like I haven't landed the commit though). this might be a good time to add it here as well. (or we can leave it for another pr)

Comment (keita-determined) [reply webui/react/src/components/ThemeToggle.module.scss:33]: Even though I dont think adding it into `make fmt` would cause any issues, probably another PR would be good.

Comment (hamidzr) [reply webui/react/src/components/ThemeToggle.module.scss:33]: yeah ofc

**Comment by keita-determined** [comment]
test fails due to shared-web. Will merge this PR and adjust code in SaaS quickly.

**Pull request merged**
